Add user to team/60

lmtjanssen opened the pull request:
fixes #60 

also the API now runs and executes queries and mutations correctly and fixes #82

lmtjanssen committed:
Made lots of changes for GraphQL's non-acceptance of Longs.

Furthermore initialised a user to be used for adding it to a team.

lmtjanssen committed:
Merge branch 'master' into addUserToTeam/60

lmtjanssen committed:
Implemented adding a user to a team including testing.

Also implemented MockUserRepository2 for testing, since I don't know how to work with MockJPARepository.

Fixed lots of bugs from previous pull request.

lfb0801 committed:
Merge branch 'master' into addUserToTeam/60

lfb0801 commented on src/main/java/com/fontys/api/service/TeamService.java:
```diff
@@ -12,32 +13,42 @@
+    }
+
+    @Transactional
+    public String addUserToTeam(Integer teamId, Integer userId) {
```
Manual testing shows that this function doesn't actually work

lmtjanssen replied:
It only works if a user and team are created, as should always be the case

lfb0801 replied:
I tested the function in the right order, but if I did it wrong, then I should have gotten a usefull exception

lfb0801 commented on src/main/resources/graphql/schema.graphqls:
```diff
@@ -0,0 +1,22 @@
+type Query {
```
position of the comments is changed when showing the schema, so it currently looks weird

lmtjanssen replied:
Agreed, looking for a way to display it nicely

lfb0801 commented on src/main/resources/graphql/schema.graphqls:
```diff
@@ -0,0 +1,22 @@
+    tournament(id:ID!):Tournament
+}
+
+type Mutation {
```
position of the comments is changed when showing the schema, so it currently looks weird

lmtjanssen replied:
Agreed, looking for a way to display it nicely

lfb0801 commented on src/main/java/com/fontys/api/entities/Team.java:
```diff
@@ -1,28 +1,36 @@
 @Data
 @Entity
+@NoArgsConstructor
 public class Team {
```
misses @AllArgsConstructor

lmtjanssen replied:
I'll add it

lfb0801 commented on src/main/java/com/fontys/api/mutations/TeamMutation.java:
```diff
@@ -19,7 +19,11 @@ public Team createTeam(String name) {
+        return teamService.deleteTeam(id);
+    }
+
+    public String addUserToTeam(Integer teamId, Integer userId) {
```
this function doesn't seem to do anything

lmtjanssen replied:
addUserToTeam or deleteTeam?

lfb0801 replied:
addUserToTeam

lfb0801 reviewed (changes requested):
implement requested changes and let's hope we never have to use mock after this. Also doesn't this pr fix #82 ?

lmtjanssen committed:
Added a better strategy for generating id's.

Added an errormessage if team or user does not exist.

Deleted the comments in the graphql scheme for better visuals in graphiql.

lmtjanssen committed:
Merge remote-tracking branch 'origin/addUserToTeam/60' into addUserToTeam/60

lmtjanssen commented:
Fixes #82 as well

larsvandriel commented on src/main/java/com/fontys/api/entities/Team.java:
```diff
@@ -1,28 +1,35 @@
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
+import javax.persistence.*;
```
I think it's beter to have specific endpoint instead of importing the whole package at once
